hyperopt/early_stop.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def no_progress_loss(iteration_stop_count=20, percent_increase=0.0):
+    """
+    Stop function that will stop after X iteration if the loss doesn't increase
+
+    Parameters
+    ----------
+    iteration_stop_count: int
+        search will stop if the loss doesn't improve after this number of iteration
+    percent_increase: float
+        allow this percentage of variation within iteration_stop_count.
+        Early stop will be triggered if the data didn't change for more than this number
+        after iteration_stop_count rounds
+    """
+    def stop_fn(trials, best_loss=None, iteration_no_progress=0):
+        new_loss = trials.trials[len(trials.trials)-1]['result']['loss']
```

Comment:
Does this criteria fail when spark trials are used in parallel? I'm getting `KeyError: 'loss'` when trying this out, I believe because this callback was executed after one trial finished, but `trials.trials[len(trials.trials)-1]` doesn't refer to that trial